core/src/main/java/hudson/security/TokenBasedRememberMeServices2.java

```diff
@@ -58,9 +62,10 @@
  * is never available in clear text.
  *
  * @author Kohsuke Kawaguchi
+ * @see TokenBasedRememberMeServices
  */
 @Restricted(NoExternalUse.class)
-public class TokenBasedRememberMeServices2 extends TokenBasedRememberMeServices {
+public class TokenBasedRememberMeServices2 extends AbstractRememberMeServices {
```

Comment:
Extending the parent of `TokenBasedRememberMeServices` because `TokenBasedRememberMeServices` itself now has too many things that we do not want and that are incompatible with `TokenBasedRememberMeServices2`; specifically, `TokenBasedRememberMeServices` now relies on an enum that only contains MD5 and SHA256 and cannot be extended to support the HmacSHA256 algorithm `TokenBasedRememberMeServices2` uses.